Source/SwiftLintBuiltInRules/Rules/Idiomatic/PreferAssetSymbolsRule.swift

```diff
@@ -0,0 +1,119 @@
+        name: "Prefer Asset Symbols",
+        description: "Prefer using asset symbols over string-based image initialization to avoid typos and enable compile-time checking",
+        rationale: """
+            UIKit.UIImage(named:) and SwiftUI.Image(_:) contain the risk of bugs due to typos. \
+            Since Xcode 15, Xcode generates codes for images in the Asset Catalog and it can avoid typos by providing compile-time checking.
+            """,
+        kind: .idiomatic,
+        minSwiftVersion: .fiveDotNine,
+        nonTriggeringExamples: [
+            // UIKit - using asset symbols
+            Example("UIImage(resource: .someImage)"),
+            Example("UIImage(systemName: \"trash\")"),
+            // SwiftUI - using asset symbols  
+            Example("Image(.someImage)"),
+            Example("Image(systemName: \"trash\")"),
+            // Dynamic strings (variables or interpolated)
+            Example("UIImage(named: imageName)"),
+            Example("UIImage(named: \"image_\\(suffix)\")"),
+            Example("Image(imageName)"),
+            Example("Image(\"image_\\(suffix)\")"),
+            // Bundle-specific initializers
+            Example("UIImage(named: \"someImage\", in: Bundle.main, compatibleWith: nil)"),
+            Example("Image(\"someImage\", bundle: Bundle.main)"),
+        ],
+        triggeringExamples: [
+            // UIKit examples
+            Example("↓UIImage(named: \"some_image\")"),
+            Example("↓UIImage(named: \"some image\")"),
+            Example("↓UIImage.init(named: \"someImage\")"),
+            // SwiftUI examples  
+            Example("↓Image(\"some_image\")"),
+            Example("↓Image(\"some image\")"),
+            Example("↓Image.init(\"someImage\")"),
+        ]
+    )
+}
+
+private extension PreferAssetSymbolsRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            let calledExpression = node.calledExpression.trimmedDescription
+            
+            // Check for UIImage(named:) calls
+            if isUIImageNamedInit(node: node, name: calledExpression) {
+                violations.append(node.positionAfterSkippingLeadingTrivia)
+            }
+            // Check for SwiftUI Image(_:) calls
+            else if isSwiftUIImageInit(node: node, name: calledExpression) {
+                violations.append(node.positionAfterSkippingLeadingTrivia)
+            }
+        }
+        
+        private func isUIImageNamedInit(node: FunctionCallExprSyntax, name: String) -> Bool {
+            // Check if this is a UIImage or UIImage.init call
+            guard uiImageInits().contains(name) else {
+                return false
+            }
+            
+            // Check if it has a "named" parameter with a string literal
+            guard let namedArgument = node.arguments.first(where: { $0.label?.text == "named" }),
```

Comment:
It must be the **first** argument not "any `named` argument anywhere in the call", right?

You can actually do `node.arguments.onlyElement`. That covers all cases in one condition (it ensure that there are no other labels).